libbpf-tools/biolatency.bpf.c

```diff
@@ -53,9 +57,15 @@ int trace_rq_start(struct request *rq, int issue)
 	u64 ts = bpf_ktime_get_ns();
 
 	if (filter_dev) {
-		struct gendisk *disk = BPF_CORE_READ(rq, rq_disk);
+		struct request_queue___x *q = (void *)BPF_CORE_READ(rq, q);
```

Comment:
I don't think this works for old kernel. See #3691 for an example.

Reply:
I referred to #3691 before creating this pull request. req->q first appears since kernel 3.0, and remains unchanged till now. So BPF_CORE_READ(rq, q) will always return a non-NULL pointer.

```
struct request {
    struct request_queue *q;
    ...
};
```

There is another way, define request__x, but it's not simple as current way: 

```
struct request___x {
	struct gendisk *disk;
} __attribute__((preserve_access_index));

```

Reply:
Okay, thanks for explanation. looks like req->q is always valid for supported kernels so we are fine here.